commit a3a9c59a683658d881aa2695be2aa875598f2712
Author: David Hildenbrand <dahi@linux.vnet.ibm.com>
Date:   Tue Oct 14 09:44:55 2014 +0200

    KVM: s390: SIGP SET PREFIX cleanup
    
    This patch cleanes up the the SIGP SET PREFIX code.
    
    A SIGP SET PREFIX irq may only be injected if the target vcpu is
    stopped. Let's move the checking code into the injection code and
    return -EBUSY if the target vcpu is not stopped.
    
    Reviewed-by: Jens Freimann <jfrei@linux.vnet.ibm.com>
    Acked-by: Cornelia Huck <cornelia.huck@de.ibm.com>
    Signed-off-by: David Hildenbrand <dahi@linux.vnet.ibm.com>
    Signed-off-by: Christian Borntraeger <borntraeger@de.ibm.com>

diff --git a/arch/s390/kvm/interrupt.c b/arch/s390/kvm/interrupt.c
index f753c0bf9604..1ba917638bba 100644
--- a/arch/s390/kvm/interrupt.c
+++ b/arch/s390/kvm/interrupt.c
@@ -1026,6 +1026,9 @@ static int __inject_set_prefix(struct kvm_vcpu *vcpu, struct kvm_s390_irq *irq)
 	trace_kvm_s390_inject_vcpu(vcpu->vcpu_id, KVM_S390_SIGP_SET_PREFIX,
 				   prefix->address, 0, 2);
 
+	if (!is_vcpu_stopped(vcpu))
+		return -EBUSY;
+
 	*prefix = irq->u.prefix;
 	set_bit(IRQ_PEND_SET_PREFIX, &li->pending_irqs);
 	return 0;